### PR TITLE
Add missing configs and methods from Sinon 16/17 

### PR DESCRIPTION
**Pull request by fatso83**
- Document sandbox.replace.usingAccessor(obj,prop,repl)
- Document assertOptions in the createSandbox call

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://sinonjs.org/releases/v17/sandbox/
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

Not totally sure what is meant by the 'header' TBH. The PR title?

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/sinon/index.d.ts
1. types/sinon/sinon-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against fef38b7a28ba42b74ea99bd8013c3058630329ec
</p>

File: types/sinon/index.d.ts [hunk 1]
```diff
@@ -1449,6 +1449,13 @@ declare namespace Sinon {
          * you might want to do:
          */
         useFakeServer: boolean | SinonFakeServer;
+        /**
+         * The assert options can help limit the amount of output produced by assert.fail
+         */
+        assertOptions: {
+            shouldLimitAssertionLogs: boolean;
+            assertionLogLimit: number;
```
Comment (43081j) [inline types/sinon/index.d.ts:1457]: are these actually optional? i noticed in your source you have defaults. so could i do `{assertionLogLimit: N}`?

**Review by 43081j — APPROVED** [review]
looks good to me 🥳  i left one question for you

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/sinon/sinon-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4cf701e76171786a516d822b85b945b8d935dce7
</p>

**Review by 43081j — APPROVED** [review]
think you might need to format the tests file but looks good to me now

**Comment by fatso83** [comment]
Ready to merge

**Pull request merged**